CHANGELOG.md

```diff
@@ -83,6 +83,9 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
     with a `seqan3::align_cfg::lower_diagonal` and `seqan3::align_cfg::upper_diagonal` instead of `seqan3::static_band`.
     It also directly exposes the lower_diagonal and upper_diagonal as public members
     ([\#1873](https://github.com/seqan/seqan3/pull/1873)).
+  * The `seqan3::align_cfg::alignment_result_capture` tag is renamed to `seqan3::align_cfg::detail::result_type` and
+    used like this: `usage: seqan3::align_cfg::detail::result_type<alignment_result_type<...>>`
```

Comment:
Maybe also add the namespace for `alignment_result_type`?

Reply:
@rrahn correctly said that this is just developer API and thus does not need a changeling entry. Changed that now!